### PR TITLE
fix centos6/amzn rpm packager

### PR DESCRIPTION
**Pull request by ehlerst**
```
rpm -qlp git/metrictank-tehlers/build/sysvinit/metrictank-0.7.0-beta1-138-gf1868cc.el6.x86_64.rpm
/etc/init.d/metrictank
/etc/raintank/metrictank.ini
/etc/raintank/storage-schemas.conf
/etc/sysconfig/metrictank
/usr/sbin/metrictank

```

### CODE REVIEW

**Comment by ehlerst** [comment]
I know upstart is supported in cent6 and amzn linux but many automatons such as chef default to the systems default init system, which in both cases is sysvinit. This causes chef to fail unless you change the service provider to Chef::Provider::Service::Upstart.

**Comment by ehlerst** [comment]
Now I'm conflicted on this change, i guess in cent6 upstart became the default however chefs provider still uses sysvinit by default on cent6.

I think the problem is, its 1/4th migrated sysvinit still being dominant. 

```
ls /etc/init/ |wc -l
19
ls /etc/init.d/ |wc -l
54
```

Going to go with upstart i guess.

**Comment by ehlerst** [comment]
After fighting with this now for a while. Upstart logging in centos6/amazon-linux is not configured by default more users will find this frustrating than useful.

**Comment by ehlerst** [comment]
For anybody wanting logs on upstart on cent6/amzn without this merge add something like this.

```
script
exec /usr/sbin/metrictank "-config=/etc/raintank/metrictank.ini"  >> /var/log/metrictank.log 2>&1 
end script

```
console log does not work, as the upstart version is too old.

**Comment by Dieterbe** [comment]
any chance centos6 has a more recent version of upstart now where logging works fine? 

should we just add those lines to scripts/config/upstart/metrictank ?

**Comment by ehlerst** [comment]
I don't think so, we probably have to add the same line. A week ago Amazon released 'amazon Linux 2' which is rhel7 compatible. It's probably still worth fixing though for those unfortunate users who can't leave el6 for whatever reason.

I'll have time to test and update this PR next week, on vacation right now!

**Comment by Dieterbe** [comment]
ok that works. happy holidays Tim!

**Comment by Dieterbe** [comment]
note that the build scripts have changed a bit in master, so will need to rebase on top of latest master.

**Comment by ehlerst** [comment]
The new upstart version:

```
rpm -qlp /home/tim/git/metrictank/build_pkg/upstart-0.6.5/metrictank-0.7.4-674-gaef00d4.el6.x86_64.rpm
/etc/init/metrictank.conf
/etc/metrictank/metrictank.ini
/etc/metrictank/storage-aggregation.conf
/etc/metrictank/storage-schemas.conf
/usr/sbin/metrictank
```

**Comment by Dieterbe** [comment]
> carbon is in bin and metrictank is in sbin, why?

no good reason at all. i created #820 for this.

thanks for your PR!

File: scripts/config/upstart-0.6.5/metrictank.conf [hunk 1]
```diff
@@ -21,5 +21,14 @@ limit nofile 102400 102400
...
 
+### NOTE: logging does not work on the older version of upstart without a redirect as explained here http://upstart.ubuntu.com/cookbook/#versions-of-upstart-older-than-1-4 
+script
+  # When loading default and sysconfig files, we use `set -a` to make
+  # all variables automatically into environment variables.
+  set -a
+  [ -r "/etc/default/metrictank" ] && . "/etc/default/metrictank"
+  [ -r "/etc/sysconfig/metrictank" ] && . "/etc/sysconfig/metrictank"
+  set +a
+  exec chroot --userspec root:root / /usr/sbin/metrictank "/etc/metrictank/metrictank.ini" >> /var/log/metrictank-stdout.log 2>> /var/log/metrictank-stderr.log
```
Comment (Dieterbe) [inline scripts/config/upstart-0.6.5/metrictank.conf]: from `metrictank -h` : 
```
  -config string
    	configuration file path (default "/etc/metrictank/metrictank.ini")

```

**Comment by Dieterbe** [comment]
thank you! (long overdue)

**Pull request merged**